analysis_driver/quality_control/genotype_validation.py

```diff
@@ -1,148 +1,107 @@
+        :param fastq_files: 1 or 2 fastqs
         :return: A pipe-separated bash command
         """
+        header = '@RG\\tID:1\\tSM:' + self.dataset.name
+        reference = cfg.query('genotype-validation', 'reference')
         bwa_bin = cfg.query('tools', 'bwa', ret_default='bwa')
+
         if len(fastq_files) == 2:
-            command_aln1 = '%s aln %s %s' % (bwa_bin, reference, fastq_files[0])
-            command_aln2 = '%s aln %s %s' % (bwa_bin, reference, fastq_files[1])
-            command_bwa = "%s sampe -r '@RG\\tID:1\\tSM:%s' %s <(%s) <(%s) %s %s" % (
-                bwa_bin, sample_name, reference, command_aln1, command_aln2, fastq_files[0], fastq_files[1]
+            aln1 = '%s aln %s %s' % (bwa_bin, reference, fastq_files[0])
+            aln2 = '%s aln %s %s' % (bwa_bin, reference, fastq_files[1])
+            command_bwa = "%s sampe -r '%s' %s <(%s) <(%s) %s %s" % (
+                bwa_bin, header, reference, aln1, aln2, fastq_files[0], fastq_files[1]
             )
         elif len(fastq_files) == 1:
-            command_aln1 = '%s aln %s %s' % (bwa_bin, reference, fastq_files[0])
-            command_bwa = "%s samse -r '@RG\\tID:1\\tSM:%s' %s <(%s) %s" % (bwa_bin, sample_name, reference,
-                                                                            command_aln1, fastq_files[0])
+            aln1 = '%s aln %s %s' % (bwa_bin, reference, fastq_files[0])
+            command_bwa = "%s samse -r '%s' %s <(%s) %s" % (bwa_bin, header, reference, aln1, fastq_files[0])
 
         else:
             raise PipelineError('Bad number of fastqs: ' + str(fastq_files))
 
-        command_samblaster = '%s --removeDups' % (cfg.query('tools', 'samblaster', ret_default='samblaster'))
-        command_samtools = '%s view -F 4 -Sb -' % (cfg.query('tools', 'samtools', ret_default='samtools'))
+        command_samblaster = '%s --removeDups' % cfg.query('tools', 'samblaster', ret_default='samblaster')
+        command_samtools = '%s view -F 4 -Sb -' % cfg.query('tools', 'samtools', ret_default='samtools')
         command_sambamba = '%s sort -t 16 -o %s /dev/stdin' % (
-            cfg.query('tools', 'sambamba', ret_default='sambamba'), expected_output_bam
+            cfg.query('tools', 'sambamba', ret_default='sambamba'), self.output_bam
         )
 
         return ' | '.join([command_bwa, command_samblaster, command_samtools, command_sambamba])
 
     def _bwa_alignment(self):
-        """
-        Run bwa alignment for all fastq files against a synthetic genome.
-        :rtype: list
-        :return list of bam file containing the reads aligned.
-        """
-        expected_bam = os.path.join(self.working_dir, self.sample_id + '_geno_val.bam')
-
-        command = self._bwa_aln(
-            self.fastq_files,
-            self.sample_id,
-            expected_bam,
-            cfg.query('genotype-validation', 'reference')
-        )
-        
-        self.dataset.start_stage('genotype_validation_bwa')
```

Comment:
GenotypeValidation no longer starts stages itself, instead letting this be done by the `Stage` superclass. If we want genotype validation to be multi-stage, then it could be segmented into a mini-pipeline.